File: practica-multilenguaje/python/palindromo.py
def es_palindromo(texto):
    texto_procesado = texto.replace(" ", "").lower()
    return texto_procesado == texto_procesado[::-1]

if __name__ == "__main__":
    entrada = input("Ingrese una palabra o frase para verificar si es palindromo:\n")
    
    if es_palindromo(entrada):
        print(f"'{entrada}' es un palindromo.")
    else:
        print(f"'{entrada}' no es un palindromo.")

File: practica-multilenguaje/java/Palindromo.java
import java.util.Scanner;

public class Palindromo {

    public static boolean esPalindromo(String texto) {
        // Eliminar espacios y convertir a minúsculas
        String textoProcesado = texto.replaceAll("\\s+", "").toLowerCase();

        int izquierda = 0;
        int derecha = textoProcesado.length() - 1;

        while (izquierda < derecha) {
            if (textoProcesado.charAt(izquierda) != textoProcesado.charAt(derecha)) {
                return false;
            }
            izquierda++;
            derecha--;
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Ingrese una palabra o frase para verificar si es palíndromo:");
        String entrada = scanner.nextLine();

        if (esPalindromo(entrada)) {
            System.out.println("'" + entrada + "' es un palíndromo.");
        } else {
            System.out.println("'" + entrada + "' no es un palíndromo.");
        }

        scanner.close();
    }
}